src/layers/icmpv6.rs

```diff
@@ -176,28 +222,87 @@ impl Layer for ICMPv6 {
             }
 
             ICMPV6_ROUTER_ADVERTISEMENT => {
-                let mut flags = RouterAdvFlags::default();
                 let cur_hop_limit = u8::from_be(bytes[4]);
-                flags.managed_address_flag = ((bytes[5] >> 7) & 0x01) == 0x01;
-                flags.other_config = ((bytes[5] >> 6) & 0x01) == 0x01;
+                let flags = RouterAdvFlags {
+                    managed_address_flag: ((bytes[5] >> 7) & 0x01) == 0x01,
+                    other_config: ((bytes[5] >> 6) & 0x01) == 0x01,
+                };
 
                 let router_lifetime: u16 = (bytes[6] as u16) << 8 | (bytes[7] as u16);
                 let reachable_time: u32 = u32::from_be_bytes(bytes[8..12].try_into().unwrap());
                 let retrans_timer: u32 = u32::from_be_bytes(bytes[12..16].try_into().unwrap());
-                decoded = 16;
+
+                let option_type: u8 = u8::from_be(bytes[16]);
```

Comment:
I believe this information is as per section 4.2 of RFC 4861. Is that correct? The way you have handled this is - as if - all the options will always be present, the RFC does not specify that all of them will always be present. This is slightly trickier to handle. The correct way to handle this is as follows - 

1. Have a separate `handle_icmpv6_options` function that would return a `Vec<ICMPV6Option>` types say (The vec can potentially be empty.) 
2. For each of the ICMPv6 types, the above function would be called.
3. Also each type should just define `Unhandled(Vec<u8>)` Option variant, this will be used when you do not have a variant supported. 

This is a bit involved.